app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,308 @@
+            supervisorScope {
+                try {
+                    updateLocalAgendaForDayFromRemote(dateTime)
+
+                    emitAll(combine(
+                        taskRepository.getTasksForDayFlow(dateTime),
+                        eventRepository.getEventsForDayFlow(dateTime),
+                        reminderRepository.getRemindersForDayFlow(dateTime)
+                    ) { tasks, events, reminders ->
+                        events + tasks + reminders
+                    })
+                } catch (e: Exception) {
+                        e.printStackTrace()
+                        // don't send error to user, just log it (silent fail is ok here)
+                    }
+                }
+        }
+    }
+
+    override fun addAgendaItems(agendaItems: List<AgendaItem>) {
+        CoroutineScope(Dispatchers.IO).launch {
+            try {
+                agendaItems.forEach { agendaItem ->
+                    when (agendaItem) {
+                        is AgendaItem.Event -> {
+                            val result = eventRepository.upsertEventLocally(agendaItem)
+                            if (result is ResultUiText.Error) {
+                                throw IllegalStateException(result.message.asStrOrNull())
+                            }
+                        }
+                        is AgendaItem.Task -> {
+                            val result = taskRepository.upsertTaskLocally(agendaItem)
+                            if (result is ResultUiText.Error) {
+                                throw IllegalStateException(result.message.asStrOrNull())
+                            }
+                        }
+                        is AgendaItem.Reminder -> {
+                            val result = reminderRepository.upsertReminderLocally(agendaItem)
+                            if (result is ResultUiText.Error) {
+                                throw IllegalStateException(result.message.asStrOrNull())
+                            }
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+                // don't send error to user, just log it (silent fail is ok here)
+            }
+        }
+    }
+
+    // Upload local changes to remote
+    override suspend fun syncAgenda(): ResultUiText<Void> {
+        val syncItems = syncRepository.getSyncItems()
+        if (syncItems.isEmpty()) {
+            return ResultUiText.Success(null)
+        }
+
+        // Sync the `Create` API calls first
+        syncItems.forEach { syncItem ->
+            when (syncItem.modificationTypeForSync) {
+                ModificationTypeForSync.Created -> {
+                    when (syncItem.agendaItemTypeForSync) {
+                        AgendaItemTypeForSync.Event -> {
+                            val event = getEvent(syncItem.agendaItemId, true) ?: return@forEach
+                            createEvent(event, true)
```

Comment:
I'd always use named parameters when possible, since for PR reviewers there's no way to know what this `true` stands for